examples/sampling/graphbolt/link_prediction.py

```diff
@@ -0,0 +1,385 @@
+    # as a 'full batch'.)
+    # 'is_train' to determine if data should be shuffled. (Shuffling is
+    # generally used only in training to improve model generalization. It's
+    # not used in validation and testing as the focus there is to evaluate
+    # performance rather than to learn from the data.)
+    # [Output]:
+    # An ItemSampler object for handling mini-batch sampling.
+    # [Role]:
+    # Initialize the ItemSampler to sample mini-batche from the dataset.
+    ############################################################################
+    item_sampler = gb.ItemSampler(
+        current_set, batch_size=args.batch_size, shuffle=is_train
+    )
+
+    ############################################################################
+    # [Input]:
+    # The 'item_sampler' object and a block conversion function based
+    # on the 'is_train'.
+    # [Output]:
+    # A Mapper object that can convert sampled data blocks.
+    # [Role]:
+    # Initialize a mapper to convert mini-batches into link blocks suitable
+    # for training or evaluation.
+    ############################################################################
+    data_block_converter = Mapper(
+        item_sampler,
+        to_link_block_train if is_train else to_link_block_eval,
+    )
+
+    ############################################################################
+    # [Input]:
+    # 'is_train' to specify the training state.
+    # [Output]:
+    # The format to be used for negative sampling. (For more details, see
+    # the documentation for the LinkPredictionEdgeFormat enum class.)
+    # [Role]:
+    # Decide the edge format for negative sampling based on whether we are
+    # in training mode.
+    ############################################################################
+    format = (
+        gb.LinkPredictionEdgeFormat.INDEPENDENT
+        if is_train
+        else gb.LinkPredictionEdgeFormat.TAIL_CONDITIONED
+    )
+
+    ############################################################################
+    # [Input]:
+    # 'data_block_converter' to convert sampled data into the required
+    # format.
+    # 'args.neg_ratio' to specify the ratio of negative to positive samples.
+    # (E.g., if neg_ratio is 1, for each positive sample there will be 1
+    # negative sample.)
+    # 'format' to specify how negative sampling should be performed.
+    # 'graph' for the overall network topology where negative sampling occurs.
+    # [Output]:
+    # A UniformNegativeSampler object that will handle the generation of
+    # negative samples for link prediction tasks.
+    # [Role]:
+    # Initialize the UniformNegativeSampler for negative sampling in link
+    # prediction.
+    ############################################################################
+    negative_sampler = gb.UniformNegativeSampler(
+        data_block_converter,
+        args.neg_ratio,
+        format,
+        graph,
+    )
```

Comment:
When the is_train is false, it seems that the set (to_link_block_eval) already contains negative edges, why do we still need negative sampler?

Reply:
if the is_train not true, subgraph sampler will directly use `data_block_convert`. Maybe we should not initialize the `negative _sampler` if in eval mode?